tests/Unit/CMCollectionMembershipEvaluationComponent.tests.ps1

```diff
@@ -0,0 +1,149 @@
+    }
+    catch [System.IO.FileNotFoundException]
+    {
+        throw 'DscResource.Test module dependency not found. Please run ".\build.ps1 -Tasks build" first.'
+    }
+
+    $script:testEnvironment = Initialize-TestEnvironment `
+        -DSCModuleName $script:dscModuleName `
+        -DSCResourceName $script:dscResourceName `
+        -ResourceType 'Mof' `
+        -TestType 'Unit'
+
+    # Import Stub function
+    $script:moduleRoot = Split-Path -Parent (Split-Path -Parent $PSScriptRoot)
+    Import-Module (Join-Path -Path $PSScriptRoot -ChildPath 'Stubs\ConfigMgrCBDscStub.psm1') -Force -WarningAction SilentlyContinue
+}
+
+function Invoke-TestCleanup
+{
+    Restore-TestEnvironment -TestEnvironment $script:testEnvironment
+}
+
+Invoke-TestSetup
+
+# Begin Testing
+try
+{
+    InModuleScope $script:dscResourceName {
+
+        Describe 'ConfigMgrCBDsc - DSC_CMCollectionMembershipEvaluationComponent\Get-TargetResource' -Tag 'Get' {
+            BeforeAll {
+                $getDiscovery = @{
+                    Props = @(
+                        @{
+                            PropertyName = 'Incremental Interval'
+                            Value        = 5
+                        }
+                    )
+                }
+
+                $getInput = @{
+                    SiteCode = 'Lab'
+                }
+
+                Mock -CommandName Import-ConfigMgrPowerShellModule
+                Mock -CommandName Set-Location
+            }
+
+            Context 'When retrieving settings' {
+                It 'Should return desired result when retrieving Collection Membership Evaluation Component settings' {
+                    Mock -CommandName Get-CMCollectionMembershipEvaluationComponent -MockWith { $getDiscovery }
+
+                    $result = Get-TargetResource @getInput
+                    $result                | Should -BeOfType System.Collections.HashTable
+                    $result.SiteCode       | Should -Be -ExpectedValue 'Lab'
+                    $result.EvaluationMins | Should -Be -ExpectedValue '5'
+                }
+            }
+        }
+
+        Describe 'ConfigMgrCBDsc - DSC_CMCollectionMembershipEvaluationComponent\Set-TargetResource' -Tag 'Set' {
+            BeforeAll {
+                $getReturn = @{
+                    SiteCode       = 'Lab'
+                    EvaluationMins = 5
+                }
+
+                $getReturnMismatch = @{
+                    SiteCode       = 'Lab'
+                    EvaluationMins = 10
+                }
+
+                Mock -CommandName Import-ConfigMgrPowerShellModule
+                Mock -CommandName Set-Location
+                Mock -CommandName Set-CMCollectionMembershipEvaluationComponent
+            }
+
+            Context 'When Set-TargetResource runs successfully' {
+                It 'Should call expected commands when changing settings' {
+                    Mock -CommandName Get-TargetResource -MockWith { $getReturn }
```

Comment:
just a note, Get-TargerResource in all resources this could be mocked at the describe level and removed from the It blocks

Reply:
Moved to describe